docs/Expo.md

```diff
@@ -0,0 +1,59 @@
+# Expo installation
+
+> This package cannot be used in the "Expo Go" app because [it requires custom native code](https://docs.expo.io/workflow/customizing/).
+
+First install the package with yarn, npm, or [`expo install`](https://docs.expo.io/workflow/expo-cli/#expo-install).
+
+```sh
+expo install react-native-health
+```
+
+After installing this npm package, add the [config plugin](https://docs.expo.io/guides/config-plugins/) to the [`plugins`](https://docs.expo.io/versions/latest/config/app/#plugins) array of your `app.json` or `app.config.js`:
+
+```json
+{
+  "expo": {
+    "plugins": ["react-native-health"]
+  }
+}
+```
+
+Next, rebuild your app as described in the ["Adding custom native code"](https://docs.expo.io/workflow/customizing/) guide.
+
+## API
+
+The plugin provides props for extra customization. Every time you change the props or plugins, you'll need to rebuild (and `prebuild`) the native app. If no extra properties are added, defaults will be used.
+
+- `healthSharePermission` (_string_): Sets the iOS `NSHealthShareUsageDescription` permission message to the `Info.plist`. Defaults to `Allow $(PRODUCT_NAME) to check health info`.
+- `healthUpdatePermission` (_string_): Sets the iOS `NSHealthUpdateUsageDescription` permission message to the `Info.plist`. Defaults to `Allow $(PRODUCT_NAME) to update health info`.
+- `isClinicalDataEnabled` (_boolean_): Adds `health-records` to the `com.apple.developer.healthkit.access` entitlement in the iOS project. Defaults to false.
+
+`app.config.js`
+
+```json
+{
+  "expo": {
+    "plugins": [
+      [
+        "react-native-health",
+        {
+          "isClinicalDataEnabled": true,
+          "healthSharePermission": "Custom health share permission",
+          "healthUpdatePermission": "Custom health update permission"
+        }
+      ]
+    ]
+  }
+}
+```
+
+## Background Processing
+
+Background processing is not currently supported by this plugin.
```

Comment:
I don't have much experience with Expo myself, but got curious about why we are not supporting Background processing via Expo?

Is there a limitation based on the native custom code required to fetch samples (as mentioned above)?

Reply:
In Expo modules, we use an AppDelegate swizzler so the package defines all of the required AppDelegate modifications itself ([example](https://github.com/expo/expo/blob/master/packages/expo-facebook/ios/EXFacebook/EXFacebookAppDelegate.m)). We could regex changes into the user's AppDelegate via the config plugin, but that seems risky. We're currently working on abstracting our swizzler in a way where other modules can implement it, then only have it applied when using Expo/unimodules.

For now, it seems best to document the limitation, and roll out support in a future update.